src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/sender/LogSenderHelper.java

```diff
@@ -120,24 +120,40 @@ public static void sendTestResult(TestCaseModel testCaseModel, BuildModel buildM
         if (testCaseModel != null) {
             if (CollectionUtils.isNotEmpty(testCaseModel.getTestResults())) {
                 List<TestCaseResultModel> testResults = testCaseModel.getTestResults();
-
-                //100 test cases at a time
-                List<List<TestCaseResultModel>> partition = Lists.partition(testResults, NUMBER_OF_TEST_CASES);
-                for(List<TestCaseResultModel> testCaseResultModels : partition){
-                    testCaseResultModels.forEach(testCase -> {
-                        if("Failed".equals(testCase.getStatus())){
-                            testCase.setErrorDetails(format(testCase.getErrorDetails()));
-                            testCase.setErrorStackTrace(format(testCase.getErrorStackTrace()));
-                        }
-                    });
-                    data.put("testResult", testCaseResultModels);
-                    LogSender.getInstance().sendLogs(pluginDescriptor.getUrl(), gson.toJson(data).getBytes()
-                            , null, pluginDescriptor.getSourceCategory());
+                // send test cases based on the size
+                List<TestCaseResultModel> toBeSent = new LinkedList<>();
+                int size = gson.toJson(data).getBytes().length;
+                for (TestCaseResultModel testCaseResultModel : testResults) {
+                    if ("Failed".equals(testCaseResultModel.getStatus())) {
+                        testCaseResultModel.setErrorDetails(format(testCaseResultModel.getErrorDetails()));
+                        testCaseResultModel.setErrorStackTrace(format(testCaseResultModel.getErrorStackTrace()));
+                    }
+                    size = size + gson.toJson(testCaseResultModel).getBytes().length;
+                    if (size > MAX_DATA_SIZE) {
+                        sendTestResultInChunksOfPreDefinedSize(buildModel, gson, pluginDescriptor, toBeSent, data);
+                        toBeSent.clear();
+                    }
+                    toBeSent.add(testCaseResultModel);
+                    data.put("testResult", toBeSent);
```

Comment:
This can be done out of the for loop as we are not recreating the toBeSent list.

Reply:
Done